### PR TITLE
Fix #648: TSLint converter doesn't handle innerReplacements

### LINKED ISSUE #648: TSLint converter doesn't handle all "fix" cases
The TSLint converter uses a regex in an attempt to ensure that all `fix` objects are wrapped in arrays (see method `TSLoader.NormalizeTSLintFixFormat`). For example, it converts
```
    "fix": {
      "innerStart": 169728,
      "innerLength": 0,
      "innerText": " "
    },
```
to
```
    "fix": [
      {
      "innerStart": 169728,
      "innerLength": 0,
      "innerText": " "
      }
    ],
```

But there are two problems:

1. The regex ignores the possibility that the `fix` object might have this structure:
```
    "fix": {
      "innerRuleName": "no-consecutive-blank-lines",
      "innerReplacements": [
        {
          "innerStart": 467,
          "innerLength": 2,
          "innerText": ""
        }
      ]
    },
```
This causes the converter to fail on the file TSLint-03.json.

**NOTE** I can't attach a `.json` file to a GitHub issue -- ask me for the failing files if you need to see them.

2. The regex (which attempts to match the `{` following `"fix":` with the next `}`) does not account for the fact that the `innerText` element might contain braces, for example:
```
    "fix": {
      "innerStart": 597,
      "innerLength": 3,
      "innerText": "{}"
    },
```
This causes the converter to fail on the file TSLint-06.json.

In any case, [JSON is not a regular language](https://cstheory.stackexchange.com/questions/3987/is-json-a-regular-language) so attempting to parse it with a regex is doomed to fail. I don't know if the subset of JSON emitted by TSLint is regular.

### PR DESCRIPTION
**Pull request by lgolding**
Issue #648 actually reported two distinct issues, and I should have filed them separately. This PR addresses item #&#8203;1 within this issue: `TSLintLoader` does not handle log files in which the `"fix"` object contains an array `"innerReplacements"`.

Item #&#8203;2 within this issue (`TSLintLoader` improperly parsed `"fix"` objects whose `"innerText"` was `"{}"`) was fixed in a previous PR. Therefore this PR closes #648 entirely.

With this fix, we can add the one functional test data file which was previously omitted because this bug caused it to fail: `TSLint-03.json`.

This concludes the addition of TSLint support to the SARIF SDK.

### CODE REVIEW

File: src/Sarif.Converters.UnitTests/TSLintLoaderTests.cs [hunk 1]
```diff
@@ -160,6 +160,73 @@ public void TSLintLoader_NormalizeLog_WrapsSingleFixInArray()
...
+                {
+                    ""endPosition"": {
+                        ""character"": 4,
+                        ""line"": 56,
+                        ""position"": 1876
+                    },
+                    ""failure"": ""trailing whitespace"",
+                    ""fix"": {
+                        ""innerRuleName"": ""no-trailing-whitespace"",
+                        ""innerReplacements"": [
```
Comment (lgolding) [inline src/Sarif.Converters.UnitTests/TSLintLoaderTests.cs]: This is the construct that the original TSLint converter didn't handle.

File: src/Sarif.Converters/TSLintLoader.cs [hunk 2]
```diff
@@ -73,8 +73,26 @@ public TSLintLog ReadLog(Stream input)
         // "fix":[{"innerStart":4429,"innerLength":0,"innerText":"\r\n"}]
         // "fix":[{"innerStart":4429,"innerLength":0,"innerText":"\r\n"},{"innerStart":4429,"innerLength":0,"innerText":"\r\n"}]
         //
+        // The following pattern also occurs, although the most recent version of the TSLint
+        // source code does not appear to support it:
+        //
+        // "fix": {
+        //   "innerRuleName": "no-trailing-whitespace",
```
Comment (boAndron) [inline src/Sarif.Converters/TSLintLoader.cs]: I might be wrong, but from the code below it looks like we're throwing innerRuleName away. Is that intended?

Comment (lgolding) [reply src/Sarif.Converters/TSLintLoader.cs]: @boAndron Yes, we don't have a slot for that.

Comment (boAndron) [reply src/Sarif.Converters/TSLintLoader.cs]: Got it, thanks!

**Pull request merged**